### PR TITLE
LPD-36537 Create DSM visualization modes object definitions from batch engine JSON

### PR DESCRIPTION
**Pull request by markocikos**
### References

- Parent Story: [LPD-27261 Create Object Definitions from JSON batch engine instead of Java](https://issues.liferay.com/browse/LPD-27261)
- An alternative solution to https://github.com/liferay-frontend/liferay-portal/pull/4393
- A followup of https://github.com/liferay-frontend/liferay-portal/pull/4301 
- A part of https://github.com/liferay-frontend/liferay-portal/pull/4414
  - part 1: https://github.com/liferay-frontend/liferay-portal/pull/4419 

### What is the goal of this PR?

We are splitting up DSM object definitions from batch engine JSON. In this PR, we are setting persistence of three visualization modes.

No need for peer review on this PR, but I will still run CI as a sanity check.

### CODE REVIEW

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by herrardo** [comment]
LGTM! :)

File: modules/apps/frontend-data-set/frontend-data-set-impl/src/main/resources/com/liferay/frontend/data/set/internal/batch/01-object-definition.batch-engine-data.json [hunk 1]
```diff
@@ -233,6 +480,59 @@
 					"type": "Clob"
 				}
 			],
+			"objectRelationships": [
+				{
+					"deletionType": "cascade",
+					"externalReferenceCode": "L_DATA_SET_TO_DATA_SET_CARDS_SECTIONS",
```
[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-impl/src/main/resources/com/liferay/frontend/data/set/internal/batch/01-object-definition.batch-engine-data.json:486]: I made this and few other minor tweaks to fully match patterns in [headless-builder-impl](https://github.com/liferay/liferay-portal/blob/master/modules/apps/headless/headless-builder/headless-builder-impl/src/main/resources/com/liferay/headless/builder/internal/batch/01-object-definition.batch-engine-data.json). I figured that leads to highest chance to pass.

File: modules/apps/frontend-data-set/frontend-data-set-impl/src/main/resources/com/liferay/frontend/data/set/internal/batch/01-object-definition.batch-engine-data.json [hunk 2]
```diff
@@ -14,6 +14,253 @@
...
+					"DBType": "String",
+					"businessType": "Text",
+					"externalReferenceCode": "COLUMN_LABEL",
+					"indexed": true,
+					"indexedAsKeyword": false,
+					"label": {
+						"en_US": "Column Label"
+					},
+					"localized": true,
+					"name": "label",
```
[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-impl/src/main/resources/com/liferay/frontend/data/set/internal/batch/01-object-definition.batch-engine-data.json:166]: Oh, nice catch! I better change this...

**Comment by markocikos** [comment]
ci:stop

**Comment by markocikos** [comment]
ci:forward

**Comment by markocikos** [comment]
ci:report:13253413

**Comment by markocikos** [comment]
ci:report:13253413

**Comment by markocikos** [comment]
ci:report:13253413

**Comment by markocikos** [comment]
ci:report:13253413

**Comment by markocikos** [comment]
ci:report:13253413

**Comment by markocikos** [comment]
ci:forward

**Comment by markocikos** [comment]
ci:report:13615541

**Comment by markocikos** [comment]
Failure in https://github.com/liferay-frontend/liferay-portal/pull/4421#issuecomment-2357547390 is definitely not related.

**Comment by markocikos** [comment]
ci:forward:force

**Pull request closed without merging**